core/utils/batch_processor.py

```
@@ -0,0 +1,347 @@
+
+Generic batch processor with per-record transaction isolation using savepoints.
+Prevents one bad record from corrupting an entire batch.
+
+Following Actor-Network Theory: Each record is treated as an independent
+actant whose enrollment in the network can fail without disrupting the
+enrollment of other actants in the batch.
+"""
+
+import logging
+import re
+from typing import List, Dict, Any, Callable, Optional
```

Ruff (0.12.2)

Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)